__tests__/api.test.js

```diff
@@ -70,6 +70,93 @@ describe("/api/articles", () => {
+      return request(app)
+        .get("/api/articles?topic=mitch")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles).toEqual(expect.any(Array));
+          expect(body.articles.length).toBeGreaterThan(0);
+          body.articles.forEach((article) => {
+            expect(article).toMatchObject({
+              article_id: expect.any(Number),
+              author: expect.any(String),
+              title: expect.any(String),
+              topic: "mitch",
+              created_at: expect.any(String),
+              votes: expect.any(Number),
+              comment_count: expect.any(Number),
+            });
+          });
+        });
+    });
+    test("GET: 200. Returns empty array if given valid search topic when no articles are on that topic", () => {
+      return request(app)
+        .get("/api/articles?topic=paper")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles).toEqual([]);
+        });
+    });
+    test("GET: 400. Sends appropriate error message if given invalid search topic", () => {
+      return request(app)
+        .get("/api/articles?topic=not_a_topic")
+        .expect(404)
+        .then(({ body }) => {
+          expect(body.msg).toBe("topic not found");
+        });
+    });
+    test("GET: 200. Sorts by column if given valid search sort_by column", () => {
+      return request(app)
+        .get("/api/articles?sort_by=title")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles).toBeSorted({ key: "title", descending: true });
+        });
+    });
+    test("GET: 200. Default sorts by date_created if given invalid search sort_by column", () => {
+      return request(app)
+        .get("/api/articles?sort_by=not_a_column")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles).toBeSorted({
+            key: "created_at",
+            descending: true,
+          });
+        });
+    });
+    test("GET: 200. Sorts by ascending if given search order of asc", () => {
+      return request(app)
+        .get("/api/articles?order=asc")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles).toBeSorted({ key: "created_at" });
+        });
+    });
+    test("GET: 200. Sorts by descending if given search order of desc", () => {
+      return request(app)
+        .get("/api/articles?order=desc")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles).toBeSorted({
+            key: "created_at",
+            descending: true,
+          });
+        });
+    });
+    test("GET: 200. Default sorts by descending if given invalid search order", () => {
+      return request(app)
+        .get("/api/articles?order=not_asc_or_desc")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles).toBeSorted({
+            key: "created_at",
+            descending: true,
+          });
+        });
+    });
```

Comment:
In general, I like the idea of doing it like this, but the user definitely should see an error/be informed there is an error when they are putting in orders or topics etc. that are incorrect.

I'd argue due to that, that we'd want to see a lot more 400/404 errors here in the testing and that the error handling might need some changing!

Reply:
Noted, thanks. Model and tests refactored to return 400/404 errors.